packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/PendingRecordingProxyApi.java

```diff
@@ -37,15 +37,16 @@ public PendingRecording asPersistentRecording(PendingRecording pigeonInstance) {
 
   @NonNull
   @Override
-  public PendingRecording withAudioEnabled(PendingRecording pigeonInstance, boolean initialMuted) {
-    if (!initialMuted
-        && ContextCompat.checkSelfPermission(
-                getPigeonRegistrar().getContext(), Manifest.permission.RECORD_AUDIO)
-            == PackageManager.PERMISSION_GRANTED) {
-      return pigeonInstance.withAudioEnabled(false);
+  public PendingRecording withAudioEnabled(PendingRecording pigeonInstance, boolean initialMuted) {    
+    boolean hasPermission = ContextCompat.checkSelfPermission(
+            getPigeonRegistrar().getContext(), Manifest.permission.RECORD_AUDIO)
+        == PackageManager.PERMISSION_GRANTED;
+
+    if (hasPermission) {      
+      return pigeonInstance.withAudioEnabled(initialMuted);
     }
 
-    return pigeonInstance.withAudioEnabled(true);
+    return pigeonInstance;
```

Comment:
it seems that even when we start the `CameraController(enableAudio:false)` when starting to record video, if no permission was given for `RECORD_AUDIO` on android, it just throws an exeption saying the permission wasn't granted.